src/main/java/org/fcrepo/spec/testsuite/crud/StateToken.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Licensed to DuraSpace under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional information
+ * regarding copyright ownership.
+ *
+ * DuraSpace licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.fcrepo.spec.testsuite.crud;
+
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertNotNull;
+
+import io.restassured.http.Header;
+import io.restassured.http.Headers;
+import io.restassured.response.Response;
+import org.fcrepo.spec.testsuite.AbstractTest;
+import org.fcrepo.spec.testsuite.TestInfo;
+import org.testng.SkipException;
+import org.testng.annotations.Test;
+
+/**
+ * @author dbernstein
+ */
+public class StateToken extends AbstractTest {
+    private static final String STATE_TOKEN = "X-State-Token";
+    private static final String IF_STATE_TOKEN = "X-If-State-Token";
+
+    /**
+     * 3.10.1-A
+     */
+    @Test(groups = {"MAY"})
+    public void stateTokenGet() {
+        final TestInfo info = setupTest("3.10.1-A",
+                                        "Implementations may include the X-State-Token header field in GET responses " +
+                                        "to provide a token representing the current state of resource. If provided, " +
+                                        "this value must change whenever the underlying state of the resource has " +
+                                        "changed.",
+                                        "https://fcrepo.github.io/fcrepo-specification/#x-state-token", ps);
+
+        final Response createResponse = createBasicContainer(uri, info.getId());
+        final String resourceUri = getLocation(createResponse);
+        final Response getResponse = doGet(resourceUri);
+        checkStateTokensSupported(getResponse);
```

Comment:
Please add newlines and a comment around the point where the test may be skipped so that it is clear that the test may terminate at this point.

Reply:
👍